### PR TITLE
Issue 2195 ox

### PR DESCRIPTION
**Pull request by oxtoacart**
@atavism This shows the changes that I made to use my new tarfs library.  That library still needs some work, but with it in place the build is super-quick.  The basic idea is that we store the entire app folder in memory as a tar archive inside of a byte array.  On startup, we create an index of filenames into that byte array, which we use to create an http.FileSystem.  I stole some of the classes from go-bindata-assetfs.

### CODE REVIEW

**Comment by oxtoacart** [comment]
To run this, go into the flashlight folder and run

```
go install && flashlight -addr localhost:10080 -role client -httpaddr localhost:16785
```

File: genassets.bash [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-go-bindata-assetfs -pkg "http" -nocompress -nomemcopy src/github.com/getlantern/ui/app/... && mv bindata_assetfs.go src/github.com/getlantern/flashlight/http
+tarfs -pkg http src/github.com/getlantern/ui/app > src/github.com/getlantern/flashlight/http/resources.go
```
Comment (oxtoacart) [inline genassets.bash:3]: This is how one embeds resources with tarfs.  You have to `go install github.com/getlantern/tarfs/tarfs` for this to work.

Comment (oxtoacart) [inline genassets.bash:3]: ^ @atavism

File: src/github.com/getlantern/flashlight/http/http.go [hunk 2]
```diff
@@ -81,11 +84,15 @@ func Start(httpAddr string) {
 	r.HandleFunc("/whitelist", WhitelistHandler)
 	r.HandleFunc("/proxy_on.pac", servePacFile)
 
-	assets := assetFS()
-	if assets != nil {
-		r.PathPrefix("/").Handler(http.FileServer(assets)).Methods("GET")
-		openUI()
+	start := time.Now()
+	fs, err := tarfs.New(Data, "../ui/app")
```
Comment (oxtoacart) [inline src/github.com/getlantern/flashlight/http/http.go:88]: @atavism This is the API for getting a filesystem.  The first parameter is a byte array (from resources.go) that contains the tar data.  The second parameter is an optional path to a local directory.  The filesystem will first search that local directory and only go to the byte array if the resource wasn't found.

**Comment by oxtoacart** [comment]
Sorry about all the extra files.  I used `gost get` to vendor the dependencies.

**Comment by atavism** [comment]
@oxtoacart I already merged this with issue-2195. I deleted the ui/app directory, and it's working great:

```
http: UI Directory does not exist stat src/github.com/getlantern/ui/app: no such file or directory
http: tarfs startup time: 3.708497ms
http: Starting UI HTTP server at localhost:16785
```

**Comment by oxtoacart** [comment]
Awesome!

Cheers,
Ox

---

"I love people who harness themselves, an ox to a heavy cart,
who pull like water buffalo, with massive patience,
who strain in the mud and the muck to move things forward,
who do what has to be done, again and again."
- Marge Piercy

**Pull request closed without merging**